src/sisl_toolbox/siesta/minimizer/_minimize.py

```diff
@@ -332,6 +343,77 @@ def run(self, *args, **kwargs):
         return _convert_optimize_result(self, opt)
 
 
+@set_module("sisl_toolbox.siesta.minimizer")
+class BADSMinimize(BaseMinimize):
+    """Bayesian Adaptive Direct Search (BADS) minimizer.
+
+    It uses the pybads package to perform the minimization.
+    """
+
+    def run(self, options={}, get_hard_bounds: Callable = lambda x: x, **kwargs):
+        from pybads.bads import BADS
+
+        bounds = self.normalize_bounds()
+        bounds = np.array(bounds)
+
+        bounds[bounds == 0] = 0.5
```

Comment:
Why is this?

What happens when one has a bounds `[0, 0.2]` then it would be changed to `[0.5, 0.2]`?

Reply:
Yes, I guess, but who has 0.2 as the maximum cutoff for the basis? I agree that this is not general, And I don't remember what happened if one of the bounds was 0, maybe something nasty :sweat_smile:

Reply:
but if you want 0.5 to be the minimum, then we should have `bounds[bounds < 0.5] = 0.5`, right?

Otherwise we should select a *sensible* minimum, and let users control everything.

Reply:
Yes clearly something should be done if there is someone who wants to use this for something other than basis optimization.

Reply:
Just read through my old yaml conf, we have to remove this, because there are soft-confinement things that uses negative bounds, so you would have to know which kind of boundary it attaches to...

This check was only done for the bads, but in principle, it should be there for all, so I'll remove it...